packages/e2e-test-utils/src/inserter.js

```diff
@@ -100,157 +142,208 @@ async function waitForInserterCloseAndContentFocus() {
-	await page.keyboard.type( searchTerm );
-	await waitForInserterSearch();
+	return await searchGlobalInserter( 'Blocks', searchTerm );
 }
 
 /**
- * Search for pattern in the global inserter
+ * Searches for a pattern via the global inserter.
  *
  * @param {string} searchTerm The text to search the inserter for.
+ * @return {Promise<ElementHandle|null>} The handle of the pattern to be
+ * inserted or null if nothing was found.
  */
 export async function searchForPattern( searchTerm ) {
-	await openGlobalBlockInserter();
-	// Select the patterns tab.
-	const tab = await page.waitForXPath(
-		'//div[contains(@class, "block-editor-inserter__tabs")]//button[.="Patterns"]'
-	);
-	await tab.click();
-	await page.waitForSelector( INSERTER_SEARCH_SELECTOR );
-	await page.focus( INSERTER_SEARCH_SELECTOR );
-	await pressKeyWithModifier( 'primary', 'a' );
-	await page.keyboard.type( searchTerm );
-	await waitForInserterSearch();
+	return await searchGlobalInserter( 'Patterns', searchTerm );
 }
 
 /**
- * Search for reusable block in the global inserter.
+ * Searches for a reusable block via the global inserter.
  *
  * @param {string} searchTerm The text to search the inserter for.
+ * @return {Promise<ElementHandle|null>} The handle of the reusable block to be
+ * inserted or null if nothing was found.
  */
 export async function searchForReusableBlock( searchTerm ) {
-	await openGlobalBlockInserter();
-
-	// The reusable blocks tab won't appear until the reusable blocks have been
-	// fetched. They aren't fetched until an inserter is used or the post
-	// already contains reusable blocks, so wait for the tab to appear.
-	await page.waitForXPath(
-		'//div[contains(@class, "block-editor-inserter__tabs")]//button[@aria-label="Reusable"]'
-	);
-
-	// Select the reusable blocks tab.
-	const tab = await page.waitForXPath(
-		'//div[contains(@class, "block-editor-inserter__tabs")]//button[@aria-label="Reusable"]'
-	);
-	await tab.click();
-	await page.waitForSelector( INSERTER_SEARCH_SELECTOR );
-	await page.focus( INSERTER_SEARCH_SELECTOR );
-	await pressKeyWithModifier( 'primary', 'a' );
-	await page.keyboard.type( searchTerm );
-	await waitForInserterSearch();
+	return await searchGlobalInserter( 'Reusable', searchTerm );
 }
 
 /**
- * Opens the inserter, searches for the given term, then selects the first
- * result that appears. It then waits briefly for the block list to update.
+ * Searches for a Block Directory block via the global inserter.
  *
  * @param {string} searchTerm The text to search the inserter for.
+ * @return {Promise<ElementHandle|null>} The handle of the Block Directory block
+ * to be inserted or null if nothing was found.
+ */
+export async function searchForBlockDirectoryBlock( searchTerm ) {
+	return await searchGlobalInserter( 'Block Directory', searchTerm );
+}
+
+/**
+ * Inserts a block matching a given search term via the global inserter.
+ *
+ * @param {string} searchTerm The text to search the blocks for.
```

Comment:
Considering that this is in the context of inserting a block, this can feel confusing. Should we change it to something like: "search term by which to find the block to insert" or something 😉 

Same for the similar `insert*` helpers.

Reply:
Good point, your suggestion sounds much better. Will update!

Reply:
Updated in https://github.com/WordPress/gutenberg/pull/46366/commits/8dd3806a0a91e0b948da2e7fd8b6cea537eb1036